lib/streamlit/navigation/page.py

```diff
@@ -206,6 +206,12 @@ def __init__(
         self._page: Path | Callable[[], None] = page
         self._title: str = title or inferred_name.replace("_", " ")
         self._icon: str = icon or inferred_icon
+
+        if self._title.strip() == "":
+            raise StreamlitAPIException(
+                "The title of the page cannot be an empty string. Please provide a title."
```

Comment:
Nit: Technically the underscore case isn't passing an empty string as the title, so this error message may be a bit confusing.